src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+
+        break;
+      case 'clear':
+        for (const removedItem in stateCopy) {
+          delete stateCopy[removedItem];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const del of action.keysToRemove) {
+          delete stateCopy[del];
+        }
+    }
+    resultArray.push({ ...stateCopy });
+  }
```

Comment:
[CODE STYLE]: switch/case should always have default case for error handling. 
Checklist 3